nix/packages/grub-efi-image.nix

```diff
@@ -0,0 +1,47 @@
+{ lib, stdenvNoCC, grub2_efi, fetchFromGitHub, git, gnulib }:
+let
+  grub_uki = grub2_efi.overrideAttrs (finalAttrs: previousAttrs: rec {
+    patches = [ ];
+    src = fetchFromGitHub {
+      owner = "GovanifY";
+      repo = "grub";
+      rev = "master";
+      sha256 = "sha256-2XhZdRCK73QJ5QIeTdqXVFinzXq36pJ/6EwnluY5rnA=";
+    };
+
+
+    prePatch = ''
+      patchShebangs autogen.sh bootstrap
+      # copy gnulib into build dir and make writable.
+      # Otherwise ./bootstrap copies the non-writable files from nix store and fails to modify them
+      cp -r ${gnulib} gnulib
+      chmod -R u+w gnulib/{build-aux,lib}
+      ./bootstrap --no-git --gnulib-srcdir=gnulib --skip-po
+      ./autogen.sh
```

Comment:
Those belong into configure or preConfigure Phase

Reply:
See above, configure is too late